test/helper.js

```diff
@@ -58,6 +62,36 @@ class FakeRoles {
   }
 }
 
+class FakePublisher {
+  constructor() {
+    this.clients = [];
```

Comment:
My approach here would be to set `this.calls = []` and then push something in each message-sending function, e.g.

```javascript
async clientUpdated({clientId}) {
  this.calls.push({method: 'clientUpdated', clientId});
}
```

Then later you can assert something about the contents of `helper.publisher.calls`.